### PR TITLE
Prometheus: Make Prometheus streaming parser as default client

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**
This makes new client (querydata) as default client for Prometheus datasource. The old client `buffered` goes behind the feature flag `prometheusBufferedClient`. 

**Why do we need this feature?**
This is another step making `querydata` a.k.a. `streaming parser` prometheus client as default. Roadmap and details here: https://github.com/grafana/grafana/issues/52926

**Who is this feature for?**
Anyone who uses Prometheus datasource

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/observability-metrics-squad/issues/53

### CODE REVIEW

**Review by kylebrandt — APPROVED** [review]

**Review by toddtreece — APPROVED** [review]

File: pkg/services/featuremgmt/registry.go [hunk 1]
```diff
@@ -208,9 +208,9 @@ var (
 			FrontendOnly: true,
 		},
 		{
-			Name:        "prometheusStreamingJSONParser",
-			Description: "Enable streaming JSON parser for Prometheus datasource",
-			State:       FeatureStateBeta,
+			Name:        "prometheusBufferedClient",
+			Description: "Use buffered (old) client as default instead of streaming JSON parser client (new)Enable for Prometheus datasource",
```
Comment (marefr) [inline pkg/services/featuremgmt/registry.go:212]: Nit. Might be wrong, but should `Enable for Prometheus datasource` be appended at the end?

**Review by marefr — APPROVED** [review]
LGTM, added a comment

File: pkg/services/featuremgmt/registry.go (hunk 1 above)
Comment (itsmylife) [reply pkg/services/featuremgmt/registry.go:212]: Yeah, that is not that much clear. I will update it and try to make it a bit more clear.

Comment (itsmylife) [reply pkg/services/featuremgmt/registry.go:212]: Oh shit I just saw the words at the end :D 
Good catch!

**Pull request merged**